src/pages-and-resources/data/thunks.js

```diff
@@ -24,9 +25,8 @@ export function fetchCourseApps(courseId) {
       }));
       dispatch(updateLoadingStatus({ courseId, status: RequestStatus.SUCCESSFUL }));
     } catch (error) {
-      // TODO: We need generic error handling in the app for when a request just fails... in other
-      // parts of the app (proctored exam settings) we show a nice message and ask the user to
-      // reload/try again later.
+      dispatch(updateApiCallStatus({ status: error.response.status }));
```

Comment:
It seems like you are only updating store with an error state, I can't seem to understand where we are setting the status for in progress. 

Re:
`const inProgress = loadingStatus === RequestStatus.IN_PROGRESS;`